rclcpp/include/rclcpp/parameter.hpp

```diff
@@ -0,0 +1,161 @@
+// Copyright 2014 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCLCPP_RCLCPP_PARAMETER_HPP_
+#define RCLCPP_RCLCPP_PARAMETER_HPP_
+
+#include <string>
+
+#include <rmw/rmw.h>
+
+#include <rclcpp/macros.hpp>
+
+
+namespace rclcpp
+{
+
+namespace parameter
+{
+  // Datatype for parameter names
+  typedef std::string param_name_t;
+
+  // Datatype for storing parameter types
+  enum ParamDataType {INT_PARAM, DOUBLE_PARAM, STRING_PARAM, BOOL_PARAM};
+
+
+  // Structure to store an arbitrary parameter with templated get/set methods
+  class ParamContainer
```

Comment:
The class needs a constructor to initialize the type id. And the type id needs to be something like "invalid" in the beginning.